src/packet.ml

```diff
@@ -84,14 +84,14 @@ let decode_header buf =
     (Cstruct.length buf >= hdr_len + (packet_id_len * arr_len) + rs)
     `Partial
   >>| fun () ->
-  let rec ack_message_id = function
-    | 0 -> []
+  let rec ack_message_id acc = function
+    | 0 -> acc
     | n ->
         let idx = pred n in
         let id = Cstruct.BE.get_uint32 buf (hdr_len + (packet_id_len * idx)) in
-        id :: ack_message_id idx
+        ack_message_id (id :: acc) idx
   in
-  let ack_message_ids = ack_message_id arr_len in
+  let ack_message_ids = ack_message_id [] arr_len in
```

Comment:
GitHub code review doesn't let me select all the relevant bits for some odd reason. How about this:
```suggestion
  let ack_message_id idx =
    Cstruct.BE.get_uint32 buf (hdr_len + (packet_id_len * idx))
  in
  let ack_message_ids = List.init arr_len ack_message_id in
```